src/main/java/com/atlassian/jira/cloud/jenkins/listeners/ConfigurationSaveableListener.java

```diff
@@ -1,41 +1,123 @@
 package com.atlassian.jira.cloud.jenkins.listeners;
 
+import com.atlassian.jira.cloud.jenkins.common.model.PluginSiteData;
+import com.atlassian.jira.cloud.jenkins.logging.PipelineLogger;
 import com.atlassian.jira.cloud.jenkins.pluginConfigApi.PluginConfigApi;
-import com.atlassian.jira.cloud.jenkins.util.IpAddressProvider;
+import com.atlassian.jira.cloud.jenkins.util.SecretRetriever;
+import com.atlassian.jira.cloud.jenkins.util.XmlUtils;
 import hudson.Extension;
 import hudson.XmlFile;
 import hudson.model.Saveable;
 import hudson.model.listeners.SaveableListener;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
 import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import static com.atlassian.jira.cloud.jenkins.util.IpAddressProvider.getIpAddress;
+
 @Extension
 public class ConfigurationSaveableListener extends SaveableListener {
+    private static final String SITE_CONFIGURATION_TAG = "atl-jsw-site-configuration";
+    private final Logger logger = Logger.getLogger(ConfigurationSaveableListener.class.getName());
 
-    private static final Logger LOGGER =
-            Logger.getLogger(ConfigurationSaveableListener.class.getName());
+    private PluginConfigApi pluginConfigApi;
+    private SecretRetriever secretRetriever;
+    private XmlUtils xmlUtils;
 
-    private transient PluginConfigApi pluginConfigApi;
+    @Inject
+    public void setSecretRetriever(final SecretRetriever secretRetriever) {
```

Comment:
We probably want to make sure the injected dependencies are not null to avoid running into NullPointerExceptions. So I say we chuck something like

```
if (secretRetriever == null) {
        throw new IllegalArgumentException("SecretRetriever cannot be null.");
    }
    ```
    
    in for these 3 injections.